src/Modules/CalcOffence.lua

```diff
@@ -3858,20 +3852,29 @@ function calcs.offence(env, actor, activeSkill)
 				end
 			end
 
-			local bleedRollAverage = bleedStacks / (bleedStacks + 1) * 100
+			-- the amount of damage each bleed does as % maximum
+			local bleedRollAverage
+			if globalOutput.BleedStackPotential > 1 then
+				-- shift damage towards top of range as only top bleeds apply
+				bleedRollAverage = globalOutput.BleedStackPotential / (globalOutput.BleedStackPotential + 1) * 100
```

Comment:
this could be improved. my thought process is the following:

the question is, when you apply bleed n (=bleedStacks) times, what are the m (=maxStacks) highest bleeds in average?
for m==1 it is n / (n + 1). for m==2, it is slightly less. the general formula is: (n - (m - 1)/2) / (n + 1).

e.g. for n==m==8 this gives 1/2 which is as expected, and pretty different to 8/9.

Reply:
globalOutput.BleedStackPotential != bleedStacks

globalOutput.BleedStackPotential = `n/m`, the average formula is then`(n/m) / (n/m + 1) = n / (m + n)`
for n=8, m=8, the average is 1/2
for n=16, m=8 the average is 3/4

This is the same formula currently in use, though variable meaning have been changed. I wasn't able to find any statistical basis for this forumla, or any other. But it has the right shape and values for the obvious cases so I accepted it. I'm interested to hear where your formula comes from.

Reply:
the mean of the k-th highest value is k/(n+1). i think one can intuitively understand that. i.e. imagine rolling a die. 

then calculate the average for the m highest k's: i.e. 
1/m * (n/(n+1) + (n-1)/(n+1) +... + (n-m+1)/(n+1) )
this simplifies to the formula i provided above.

i also wrote a little program to test:
i.e. your formula for n=16, m=8 gives 2/3~=0.66667, my gives 12.5/17~=0.73529

```
my program:
    public static void main(String[] args) {
        int n = 16;
        int m = 8;
        int tries = 10000000;
        Random r = new Random();
        double sumTries = 0;
        double[] bleedattempts = new double[n];
        for (int i = 0; i < tries; i++) {
            for (int j = 0; j < n; j++) {
                bleedattempts[j] = r.nextDouble();
            }
            Arrays.sort(bleedattempts);
            double sumHighest = 0;
            for (int j = 0; j < m; j++) {
                sumHighest += bleedattempts[n - m + j];
            }
            sumHighest /= m;
            sumTries += sumHighest;
        }
        sumTries /= tries;
        System.out.println("avg of " + tries + " tries of average of " + m + " highest bleeds of " + n + ": " + sumTries);
        System.out.println("expected: " + (n - (m - 1.0) / 2) / (n + 1));
    }

output:
avg of 10000000 tries of average of 8 highest bleeds of 16: 0.7352883552110671
expected: 0.7352941176470589```

Reply:
This formula seems slightly inaccurate when when `n` is a decimal instead of a whole number, but much more accurate than previously. I've integrated it. Thanks for the simulate code.